misc/lldb_rb/utils.py

```diff
@@ -249,7 +249,224 @@ def inspect(self, val):
+                elif nd_type == self.ruby_globals["NODE_TRUE"]:
+                    self._append_command_output("p *(struct RNode_TRUE *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_FALSE"]:
+                    self._append_command_output("p *(struct RNode_FALSE *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_ERRINFO"]:
+                    self._append_command_output("p *(struct RNode_ERRINFO *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_DEFINED"]:
+                    self._append_command_output("p *(struct RNode_DEFINED *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_POSTEXE"]:
+                    self._append_command_output("p *(struct RNode_POSTEXE *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_DSYM"]:
+                    self._append_command_output("p *(struct RNode_DSYM *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_ATTRASGN"]:
+                    self._append_command_output("p *(struct RNode_ATTRASGN *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_LAMBDA"]:
+                    self._append_command_output("p *(struct RNode_LAMBDA *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_ARYPTN"]:
+                    self._append_command_output("p *(struct RNode_ARYPTN *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_HSHPTN"]:
+                    self._append_command_output("p *(struct RNode_HSHPTN *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_FNDPTN"]:
+                    self._append_command_output("p *(struct RNode_FNDPTN *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_ERROR"]:
+                    self._append_command_output("p *(struct RNode_ERROR *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_RIPPER"]:
+                    self._append_command_output("p *(struct RNode_RIPPER *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_RIPPER_VALUES"]:
+                    self._append_command_output("p *(struct RNode_RIPPER_VALUES *) %0#x" % val.GetValueAsUnsigned())
+
+                # Internal nodes
+                elif nd_type == self.ruby_globals["NODE_DEF_TEMP"]:
+                    self._append_command_output("p *(struct RNode_DEF_TEMP *) %0#x" % val.GetValueAsUnsigned())
+                elif nd_type == self.ruby_globals["NODE_EXITS"]:
+                    self._append_command_output("p *(struct RNode_EXITS *) %0#x" % val.GetValueAsUnsigned())
```

Comment:
I suspect these symbols do not appear in `self.ruby_globals`.

Reply:
Oh, thank you.
One idea is moving `enum internal_node_type` into `internal/ruby_parser.h` and defining `enum internal_node_type     internal_node_type;` in `debug.c`.
Or removing internal nodes from `utils.py`.
At the moment, I don't have strong opinion. Do you have any preference?

Reply:
For the time being, how about ignoring these internal nodes?
Regarding `NODE_DEF_TEMP` at least, I'm planning to remove cast it to `NODE`.

Reply:
Make sense, updated to remove internal nodes.